app/src/test/java/com/example/parkingsystem/mvp/presenter/ParkingSpaceReservationPresenterTest.java

```diff
@@ -0,0 +1,191 @@
+        reservation.setParkingSpace(PARKING_SPACE_INT);
+        return reservation;
+    }
+
+    @Test
+    public void onFirstButtonParkingSpaceReservationPickerPressed() {
+        when(view.getButtonPickerStart()).thenReturn(true);
+        presenter.onButtonParkingSpaceReservationPickerPressed(datePickerListener);
+        verify(view).showDatePickerDialog(datePickerListener);
+    }
+
+    @Test
+    public void onSecondButtonParkingSpaceReservationPickerPressed() {
+        when(view.getButtonPickerStart()).thenReturn(false);
+        presenter.onButtonParkingSpaceReservationPickerPressed(datePickerListener);
+        verify(view).showDatePickerDialog(datePickerListener);
+    }
+
+    @Test
+    public void onMissingDateStartTest() {
+        when(view.getParkingSpace()).thenReturn(PARKING_SPACE_STRING);
+        when(view.getSecurityCode()).thenReturn(SECURITY_CODE_STRING);
+        presenter.onButtonParkingSpaceReservationSavePressed();
+        verify(view).showReleasedPastReservations(ZERO_INT);
+        verify(view).showMissingDateStart();
+    }
+
+    @Test
+    public void onMissingTimeStartTest() {
+        when(view.getParkingSpace()).thenReturn(PARKING_SPACE_STRING);
+        when(view.getSecurityCode()).thenReturn(SECURITY_CODE_STRING);
+        model.setDateStart(createDateCalendar(YEAR_START, MONTH_START, DAY_START));
+        presenter.onButtonParkingSpaceReservationSavePressed();
+        verify(view).showReleasedPastReservations(ZERO_INT);
+        verify(view).showMissingTimeStart();
+    }
+
+    @Test
+    public void onMissingDateEndTest() {
+        when(view.getParkingSpace()).thenReturn(PARKING_SPACE_STRING);
+        when(view.getSecurityCode()).thenReturn(SECURITY_CODE_STRING);
+        model.setDateStart(createDateCalendar(YEAR_START, MONTH_START, DAY_START));
+        model.setTimeStart(createTimeCalendar(HOUR_START, MINUTE_START));
+        presenter.onButtonParkingSpaceReservationSavePressed();
+        verify(view).showReleasedPastReservations(ZERO_INT);
+        verify(view).showMissingDateEnd();
+    }
+
+    @Test
+    public void onMissingTimeEndTest() {
+        when(view.getParkingSpace()).thenReturn(PARKING_SPACE_STRING);
+        when(view.getSecurityCode()).thenReturn(SECURITY_CODE_STRING);
+        model.setDateStart(createDateCalendar(YEAR_START, MONTH_START, DAY_START));
+        model.setTimeStart(createTimeCalendar(HOUR_START, MINUTE_START));
+        model.setDateEnd(createDateCalendar(YEAR_END, MONTH_END, DAY_END));
+        presenter.onButtonParkingSpaceReservationSavePressed();
+        verify(view).showReleasedPastReservations(ZERO_INT);
+        verify(view).showMissingTimeEnd();
+    }
+
+    @Test
+    public void onMissingParkingSpace() {
+        when(view.getParkingSpace()).thenReturn(EMPTY_STRING);
+        when(view.getSecurityCode()).thenReturn(SECURITY_CODE_STRING);
+        model.setDateStart(createDateCalendar(YEAR_START, MONTH_START, DAY_START));
```

Comment:
you should test the model throw the presenter so you should not call the model directly. Please try to set these values from de presenter

Reply:
Done!